test/e2e/tests/chain-interactions.spec.js

```diff
@@ -48,6 +48,14 @@ describe('Chain Interactions', function () {
 
         // approve add chain, cancel switch chain
         await driver.clickElement({ text: 'Approve', tag: 'button' });
+
+        await driver.delay(1000);
+        windowHandles = await driver.getAllWindowHandles();
```

Comment:
Why was this additional pause and request for window handles required? I can see that the webdriver was already pointing at the notification popup, but it's not clear to me what happens here.

Is the popup closing and re-opening again?

Reply:
window id is changing, approve add network has different ID than confirm switch network. Added pause to catch this change, witouth these changes it was failing with `window not found`

Reply:
Ok, that implies that the window is closing and re-opening.

I'm having trouble understanding how your changes relate to this though. You didn't adjust the switch network steps at all.

Perhaps some part of your changes is causing the window to close sooner than it had before? Ideally it would stay open long enough for the switch network prompt to appear.

Reply:
My change makes it stay open longer I think as I moved validation and adding network to after user approves. Maybe that is why window was missing in test, because previously new window would have been already opened by then?
Let me know if you would like me to make some changes please.

Reply:
I don't have any insight to share on the actual issue, just noting that it seems optimistic with a 1s window here. Starvation scenarios are not that rare.

Reply:
We should definitely try to prevent the window from closing. This would be a substantial UX degradation, and would make these e2e tests flaky.